ui/src/components/pipelines/pipeline_designer/context_bar/DebugView.js

```diff
@@ -0,0 +1,161 @@
+import React, { Component } from "react";
+import { LifeBuoy } from "react-feather";
+import * as YAML from "json-to-pretty-yaml";
+import { Button } from "react-bootstrap";
+import { Prism as SyntaxHighlighter } from "react-syntax-highlighter";
+
+class DebugView extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      tab: "error",
+    };
+    this.goToTab = this.goToTab.bind(this);
+  }
+
+  goToTab(event) {
+    event.preventDefault();
+
+    this.setState({
+      tab: event.target.dataset.tab,
+    });
+  }
+
+  render() {
+    const record = this.props.record;
+    const pipeline = this.props.pipeline;
+    const error = record.metadata["error"];
+
+    let debugContent = "Not available.";
+
+    if (this.state.tab === "error") {
+      debugContent = JSON.stringify(error, null, 2);
+    } else if (this.state.tab === "record") {
+      let originalRecord = JSON.parse(JSON.stringify(record));
+      delete originalRecord.metadata["error"];
+      delete originalRecord.id;
+      debugContent = JSON.stringify(originalRecord, null, 2);
+    } else if (this.state.tab === "transform") {
+      const match = /steps\[(\d+)\]\.fields\[(.+)\]/g.exec(
```

Comment:
Can you leave a comment, what this should match and what not?

Reply:
Thanks for raising this point. I hope the added comment is helpful?